stories/01-tutorial/05-Register-the-Component-in-WordPress.stories.mdx

```diff
@@ -0,0 +1,154 @@
+
+`/your-theme/components/donation-call-to-action/component.json`
+```json
+{
+  "name": "tutorial/donation-call-to-action",
+  "config": {
+    "title": {
+      "default": "",
+      "type": "string"
+    },
+    "subtitle": {
+      "default": "",
+      "type": "string"
+    },
+    "ctaText": {
+      "default": "",
+      "type": "string"
+    },
+    "link": {
+      "default": "",
+      "type": "string"
+    }
+  }
+}
+```
+
+## Create `component.php`
+Now that we have described our component, we can register it in PHP, and add config and children callbacks.
+
+`/your-theme/components/donation-call-to-action/component.php`
+```php
+<?php
+/**
+ * Donation Call To Action.
+ *
+ * @package Your_Theme
+ */
+
+namespace Your_Theme;
+
+use WP_Irving\Components;
+
+/**
+ * Register the component.
+ */
+Components\register_component_from_config( __DIR__ . '/component' );
+```
+
+## Update `single.json` to set your new values
+```json
+{
+  "name": "tutorial/donation-call-to-action",
+  "config": {
+    "title": "We need your support!",
+    "subtitle": "This tutorial was made possible by contributions to your PBS station from viewers like you. Thank you!",
+    "cta_text": "Learn More",
+    "link": "/donate/"
+  }
+}
+```
+
+## Update the React component to use the props
+Now that the component is passing our values from the backend, we can add those props to the React component.
+
+```js
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  standardPropTypes,
+  getStandardDefaultProps,
+} from '@irvingjs/styled/types/propTypes';
+import useStandardProps from '@irvingjs/styled/hooks/useStandardProps';
+import * as defaultStyles from './themes/default';
+
+const DonationCallToAction = (props) => {
+  const {
+    theme,
+    title,
+    subtitle,
+    ctaText,
+    link,
+  } = props;
+  const standardProps = useStandardProps(props);
+  const {
+    CTAButton,
+    Icon,
+    Wrapper,
+  } = theme;
+
+  return (
+    <Wrapper {...standardProps} maxWidth="sm">
+      <h3>{title}</h3>
+      <p>{subtitle}</p>
+      <CTAButton href={link}>{ctaText}<Icon /></CTAButton>
+    </Wrapper>
+  );
+};
+
+DonationCallToAction.defaultProps = {
+  ...getStandardDefaultProps(),
+  theme: defaultStyles,
+  title: '',
+  subtitle: '',
+  ctaText: '',
+  link: '',
+};
+
+DonationCallToAction.propTypes = {
+  ...standardPropTypes,
+  theme: PropTypes.shape({
+    CTAButton: PropTypes.elementType.isRequired,
+    Icon: PropTypes.elementType.isRequired,
+    Wrapper: PropTypes.elementType.isRequired,
+  }),
+  title: PropTypes.string,
+  subtitle: PropTypes.string,
+  ctaText: PropTypes.string,
+  link: PropTypes.string,
+};
+
+const themeMap = {
+  default: defaultStyles,
+};
+
+export {
+  DonationCallToAction as Component,
+  themeMap,
+};
+
+export default DonationCallToAction;
+```
+
+<img src="/assets/img/tutorial/04/finished.png" style={{maxWidth: "600px", margin: "auto", width: "100%", display: "flex"}} />
+
+## That's It!
```

Comment:
Seems like we need one more step here that explains the config/child callbacks, otherwise there really isn't much value in passing static data from the CMS to the front end.

Reply:
You could use a component like this in multiple places, and the static data from the layout could be modified depending on the context, so I wanted to demonstrate this.

Agreed we need to go into config & child callbacks, but I need to draw the line somewhere to get this merged. It'll be added in a followup PR for sure.